c/tskit/tables.c

```diff
@@ -7957,6 +7957,203 @@ tsk_table_collection_subset(
     return ret;
 }
 
+int TSK_WARN_UNUSED
+tsk_table_collection_union(tsk_table_collection_t *self, tsk_table_collection_t *other,
+    tsk_id_t *nodes, tsk_size_t num_nodes)
```

Comment:
how about we call this `node_map` instead of `nodes`? to be more suggestive of what it does?

Reply:
Gee, we could just use the input for `node_map` below, except then we'd be modifying the input (bad). Maybe we call the input `node_mapping` and the internal variables `X_map`? Or something?